textpipe/doc.py

```diff
@@ -608,3 +604,28 @@ def summary(self):
         []
         """
         return self.generate_textrank_summary()
+
+    @functools.lru_cache()
```

Comment:
I am not sure if we need to cache this. `self.sents` should be fast and if not a cache there maybe makes more sense?